Structured Data 2021 Markdown (#2466)
HTTPArchive/almanac.httparchive.org: src/content/en/2021/structured-data.md

---
#See https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Authors'-Guide#metadata-to-add-at-the-top-of-your-chapters
title: Structured Data
description: Structured Data chapter of the 2021 Web Almanac covering adoption and impact of schema.org, RDFa, Microdata and more.
authors: [jonoalderson, cyberandy]
reviewers: [vdwijngaert, philbarker]
analysts: [GregBrimble]
translators: [tunetheweb]
editors: [jvandriel, JasmineDWillson]
results: https://docs.google.com/spreadsheets/d/1uEA217YjpX1xdRVBnIz9ekjpjIrpGIh5GwUxvnzJNig/edit?usp=sharing
jonoalderson_bio: Jono Alderson is a digital strategist, marketing technologist, and full stack developer. He enjoys dabbling with website performance, technical SEO, schema.org and all things structured data.
cyberandy_bio: Andrea Volpini is the CEO of WordLift, and is currently focusing on the semantic web, SEO and artificial intelligence.

> featured_quote: Describing our content using structured data enables machines to treat webpages and websites as databases. That creates a wealth of possibilities for business, technology, and society.

Reviewer: Please fill in featured stats below this.

Reviewer 2: Done!

  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vT3kZ1Ys-9tId-WBa_8muMzrTAu1Ad5TXYgkopXMmBVc1xmd2N_4PZIpEZEOqRL7baymle0kHzaC6KY/pubchart?oid=1536454005&format=interactive",
  sheets_gid="1437193020",
  sql_file="facebook_types.sql"
  )
}}

Of all of the Facebook tags that we detected, there are only three tags with significant adoption.

Those are `fb:app_id`, `fb:admins`, and `fb:pages`; which we found on 6.06%, 2.63% and 0.86% of pages respectively.

These tags are used to explicitly relate a webpage to a Facebook Page/Brand, or to grant permissions to a user (or users)  who administrates those profiles.

Anecdotally, it's unclear how well these are supported by Facebook. The platform has gone through radical changes over the past few years, and their technical documentation hasn't been well-maintained. However, many content management systems, templates and best practice guides—as well as some of Facebook's debugging tools—still include and make reference to them.

> ##### Microformats (and microformats2)

Reviewer: Might wanna consider dropping the brackets:

```suggestion
##### Microformats and microformats2
```

As looks a bit odd in the Table of Contents at the moment:

![image](https://user-images.githubusercontent.com/10931297/141343508-c401008c-f513-4ee5-a2cb-e207258f8098.png)

Specifically, we've identified and extracted structured data relating to:

* [Schema.org](http://schema.org/)
* <a hreflang="en" href="https://www.dublincore.org/specifications/dublin-core/">Dublin core</a>
* Meta tags used by social networks:
    * <a hreflang="en" href="https://ogp.me/">Open Graph</a>
    * <a hreflang="en" href="https://developer.twitter.com/en/docs/twitter-for-websites/cards/guides/getting-started">Twitter</a>
    * <a hreflang="en" href="https://developers.facebook.com/docs/sharing/webmasters/">Facebook</a>
* [Microformats](http://microformats.org/) (and <a hreflang="en" href="https://microformats.org/wiki/microformats2">microformats2</a>)
* [RDFa](https://en.wikipedia.org/wiki/RDFa), [Microdata](https://en.wikipedia.org/wiki/Microdata_(HTML)) and <a hreflang="en" href="https://json-ld.org/">JSON-LD</a>

Collectively, these provide a broad overview of different use-cases and scenarios; and include both legacy standards and modern approaches (e.g., microformats vs JSON-LD).

Before we explore specific usage across the various structured data types, we should briefly explore some caveats.

### Data caveats

> #### 1. The influence of Content Management Systems

Reviewer: Headings should have at least one sentence and not be empty (so called "Stacked headings", which are used to group headings but don't have any content themselves). "A heading is like a chapter title; it must have at least a sentence of information below it. Stacked headings can indicate inefficient organization of information."

  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vT3kZ1Ys-9tId-WBa_8muMzrTAu1Ad5TXYgkopXMmBVc1xmd2N_4PZIpEZEOqRL7baymle0kHzaC6KY/pubchart?oid=1063480738&format=interactive",
  sheets_gid="379443022",
  width=600,
  height=504,
  sql_file="rdfa_type_ofs.sql"
  )
}}

When we evaluate the types of RDFa, we can see that the `foaf:image` syntax is present on far more pages than any other type—on upwards of 0.86% of all pages in our data set. Whilst that may seem like a small proportion, it represents over ~65,000 pages, and over 60% of the total RDFa markup that we discovered.

Beyond this outlier, the use of RDFa diminishes and fragments considerably, though there are still some interesting discoveries to explore.

##### On FOAF

> [FOAF](http://xmlns.com/foaf/spec/) is a linked data dictionary of people-related terms, created in the early-2000s. It can be used to describing _people_, _groups_ and _documents_.

Reviewer: Does FOAF stand for something? Should we mention that?

Reviewer 2: Clarified.

> FOAF uses W3C's RDF syntax and in its <a hreflang="en" href="https://web.archive.org/web/20140331104046/http://www.foaf-project.org/original-intro">original introduction</a> was explained as follows:

Reviewer: RDF or RDFa?

```suggestion
FOAF uses W3C's RDFa syntax and in its <a hreflang="en" href="https://web.archive.org/web/20140331104046/http://www.foaf-project.org/original-intro">original introduction</a> was explained as follows:
```

> <p class="note">"Consider a Web of inter-related home pages, each describing things of interest to a group of friends. Each new home page that appears on the Web tells the world something new, providing factoids and gossip that make the Web a mine of disconnected snippets of information. FOAF provides a way to make sense of all this."</p>

Reviewer: What is this a quote from?

Reviewer 2: Clarified

Anecdotally, we can attribute a prominence of `foaf` markup in our results to sites running on older versions of the Drupal CMS, which historically added `typeof="foaf:image"` and `foaf:document` markup to its HTML by default.

##### On other notable RDFa findings

As well as FOAF properties, various other standards and syntaxes show up in our list.

Notably, we can see several `sioc` properties, such as `sioc:item` (0.24% of pages) and `sioc:useraccount` (0.03% of pages). <a hreflang="en" href="https://www.w3.org/Submission/sioc-spec/">SIOC</a> is a standard designed to describe structured data relating to _online communities_, such as message boards, forums, wikis and blogs.

> We can also see a <a hreflang="en" href="https://www.w3.org/TR/skos-primer/">SKOS</a> property—`skos:concept`—on 0.04% of pages. SKOS is another standard, which aims to provide a way of describing taxonomies and classifications (e.g., tags, data sets, and so on).

Reviewer: Is SKOS an acronym for something?

Reviewer 2: Clarified

#### Dublin Core

> <a hreflang="en" href="https://dublincore.org/">Dublin Core</a> is a vocabulary interoperable with linked data standards that was originally conceived in Dublin, Ohio in 1995 at an OCLC (Online Computer Library Center) and NCSA (National Center for Supercomputing Applications) workshop.

Reviewer: Should we add links to OCLC and NCSA?

```suggestion
<a hreflang="en" href="https://dublincore.org/">Dublin Core</a> is a vocabulary interoperable with linked data standards that was originally conceived in Dublin, Ohio in 1995 at an 'Online Computer Library Center](https://www.oclc.org/en/home.html) (OCLC) and <a hreflang="en" href="https://www.ncsa.illinois.edu/">National Center for Supercomputing Applications</a> (NCSA) workshop.
```

  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vT3kZ1Ys-9tId-WBa_8muMzrTAu1Ad5TXYgkopXMmBVc1xmd2N_4PZIpEZEOqRL7baymle0kHzaC6KY/pubchart?oid=1985156978&format=interactive",
  sheets_gid="358057466",
  width=600,
  height=574,
  sql_file="dublin_core_types.sql"
  )
}}

That the most popular attribute type is `dc:title` (on 0.70% of pages) comes as no surprise; but it is interesting to see that `dc:language` is next (above common descriptors like _description_, _subject_ and _publisher_) with a penetration of 0.49%. This makes sense, when you consider that Dublin Core is often used in multilingual metadata management systems.

It's also interesting to see the relatively prominent appearance of `dc:relation` (on 0.16% of pages)—an attribute that is capable of expressing relationships between different concepts.

While it might seem to many that Schema.org is predominant in the context of SEO, the role of DC remains pivotal because of its broad interpretation of concepts and its deep roots in the _linked open data movement_.

#### Social metadata

> ##### Open Graph

Reviewer: Headings should have at least one sentence and not be empty (so called "Stacked headings", which are used to group headings but don't have any content themselves). "A heading is like a chapter title; it must have at least a sentence of information below it. Stacked headings can indicate inefficient organization of information."

Anecdotally, much of the adoption we see of modern standards like schema.org (particularly via JSON-LD) appears to be motivated by organizations and individuals who wish to take advantage of search engines' support (and rewards) for providing data about their pages and content. But outside of this, there's a rich landscape of people who use structured data to enrich their pages for other reasons. They describe their websites and content so that they can integrate with other systems, so that they can better understand content, or in order to facilitate _others_ to tell _their own_ stories and build their own products.

A web made of deeply connected, structured data which powers a more integrated world has long been a science-fiction dream. But perhaps, not for much longer. As these standards continue to evolve, and their adoption continues to grow, we pave a road towards an exciting future.

### Future years

As this is the first time this chapter has been included in the Web Almanac, we don't have any data we can use to explore change over time.

> In future years, we'll be able to assess how the distribution, types, and formats of structured data have changed across the web. This may provide valuable insight into growing/declining technologies, strategies, and opportunities.

Reviewer: Maybe rephrase the first sentence to be more positive?

Something along the lines of "In future years we hope to be able to continue this analysis the analysis started here and map the evolution of structured data usage over time." which sounds more like this is an opportunity, than an excuse or something we've missed or an unable to cover in this chapter?

Reviewer 2: Done
